src/availability/data_source.rs

```diff
@@ -15,51 +15,105 @@ use super::query_data::{
     TransactionIndex,
 };
 use crate::{Block, Deltas, Resolvable};
+use async_trait::async_trait;
 use futures::stream::Stream;
 use hotshot_types::traits::{
     node_implementation::{NodeImplementation, NodeType},
     signature_key::EncodedPublicKey,
 };
+use serde::{Deserialize, Serialize};
+use snafu::Snafu;
 use std::error::Error;
-use std::fmt::Debug;
+use std::fmt::{self, Debug, Display, Formatter};
+use std::ops::RangeBounds;
 
+#[derive(Clone, Copy, Debug, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub enum ResourceId<H> {
+    Number(usize),
+    Hash(H),
```

Comment:
Does it make sense to constrain this as the Id of something Committable?

Reply:
ahh yeah that's annoying, ok sounds good

Reply:
I thought about this more and I kind of convinced myself to like it. We can do `ResourceId<T: Committable>` with `Hash(Commitment<T>)`, which enforces the semantics we want, and I just think `ResourceId<Block>` makes more sense to read than `ResourceId<BlockHash>`. Plus I can also do `From<Commitment<T>> for ResourceId<T>`. I think this is a nice enough ergonomics improvement to justify adding `T: Committable` in a few more places